### PR TITLE
feat(core): global search box - search on enter

### PR DESCRIPTION
**Pull request by ranihorev**
Initiate search on enter. If an exact app match was found, open the app directly.
Bonus: replaced the deprecated "which" with "key"

### CODE REVIEW

**Review by caseyhebebrand — APPROVED** [review]

**Comment by christopherthielen** [comment]
I like this, but I have some suggestions about the behavior.  The global search already has keyboard navigation and I think it would be best to integrate with that UX.

1) Add visual cue w.r.t. what the enter key will do.  Pre-select the application using the keyboard navigation code.

2) Add visual cue w.r.t. redirecting to `/search?q=notanapp`.  Pre-select the "show all" button if no application is an exact match.

3) Redirecting to `/search?q=foo` when the search results are loaded is jarring/surprising.  (I suspect using the keyboard nav code will address this this)

4) After navigating to the app, unfocus the search (I suspect using the keyboard nav code will address this this too)

**Comment by christopherthielen** [comment]
I see a couple other UX improvements you could make as long as you're in there (if you want):
1) The keyboard nav highlight color is really low contrast
2) Somebody (it was meeeee) made a poor choice... all the "show all" links do the exact same thing.  Maybe that should be a single link on the top of the results.

**Comment by christopherthielen** [comment]
3) sometimes (it's difficult to reproduce) hitting enter submits the form.  We could add an `onSubmit` to the `<form>` to block submittal

File: app/scripts/modules/core/src/search/global/GlobalSearch.tsx [hunk 1]
```diff
@@ -119,32 +120,41 @@ export class GlobalSearch extends React.Component<{}, IGlobalSearchState> {
...
       }
-    } else if (which === 13) {
-      // enter
-      // do not submit the form and reload the page
+    } else if (key === 'Enter') {
+      const { $state } = ReactInjector;
+      if (this.state.categories) {
+        const matchingQueryResult = findMatchingApplicationResultToQuery(this.state.categories, this.state.query);
+        // eslint-disable-next-line no-console
+        console.log('matching', matchingQueryResult);
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/search/global/GlobalSearch.tsx:147]:
```suggestion

```

File: app/scripts/modules/core/src/search/global/GlobalSearchResults.tsx [hunk 2]
```diff
@@ -36,11 +33,15 @@ export const GlobalSearchResults = ({
     );
   }
 
+  const matchingApp = findMatchingApplicationResultToQuery(categories, query);
+  // eslint-disable-next-line no-console
+  console.log(matchingApp);
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/search/global/GlobalSearchResults.tsx:38]:
```suggestion
```

**Review by christopherthielen — APPROVED** [review]

**Pull request merged**